packages/pyright-internal/src/languageServerBase.ts

```diff
@@ -598,25 +603,47 @@ export abstract class LanguageServerBase implements LanguageServerInterface {
                     await this.updateSettingsForWorkspace(newWorkspace);
                 });
             });
+
+            this._setupFileWatcher();
+        }
+    }
+
+    private _setupFileWatcher() {
+        if (!this.client.hasWatchFileCapability) {
```

Comment:
Now we rely on the client's filewatching

Reply:
But clients don't provide file watching outside of the workspace. What about site-packages?

Reply:
@erictraut see my other comment (https://github.com/microsoft/pyright/pull/3298#discussion_r844260706) unfortunately, I don't believe it has worked for long time.

Reply:
If it never worked, then I'm OK removing it. I haven't received any bug reports from non-VSCode client maintainers, so I'll conclude that this wasn't a problem.